gdx/src/com/badlogic/gdx/graphics/glutils/GLVersion.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.glutils;
+
+import com.badlogic.gdx.Application;
+import com.badlogic.gdx.Gdx;
+
+import java.text.NumberFormat;
+import java.text.ParseException;
+
+public class GLVersion {
+
+	private final int majorVersion;
+	private final int minorVersion;
+	private final int releaseVersion;
+
+	private final String vendorString;
+	private final String rendererString;
+
+	private final Type type;
+
+	public GLVersion (Application.ApplicationType appType, String versionString, String vendorString, String rendererString) {
+		if (appType == Application.ApplicationType.Android) this.type = Type.GLES;
+		else if (appType == Application.ApplicationType.iOS) this.type = Type.GLES;
+		else if (appType == Application.ApplicationType.Desktop) this.type = Type.OpenGL;
+		else if (appType == Application.ApplicationType.Applet) this.type = Type.OpenGL;
+		else if (appType == Application.ApplicationType.WebGL) this.type = Type.WebGL;
+		else this.type = Type.NONE;
+
+		if (type == Type.GLES) {
+			//OpenGL<space>ES<space><version number><space><vendor-specific information>.
+			String[] versionSplit = versionString.split(" ")[2].split("\\.", 2);
+			majorVersion = parseInt(versionSplit[0], 2);
+			minorVersion = versionSplit.length < 2 ? 0 : parseInt(versionSplit[1], 0);
+			releaseVersion = 0;
+		} else if (type == Type.WebGL) {
+			//WebGL<space><version number><space><vendor-specific information>
+			String[] versionSplit = versionString.split(" ")[1].split("\\.", 2);
+			majorVersion = parseInt(versionSplit[0], 2);
+			minorVersion = versionSplit.length < 2 ? 0 : parseInt(versionSplit[1], 0);
+			releaseVersion = 0;
+		} else if (type == Type.OpenGL) {
+			//<version number><space><vendor-specific information>
+			String[] versionSplit = versionString.split(" ")[0].split("\\.", 3);
+			majorVersion = parseInt(versionSplit[0], 2);
+			minorVersion = versionSplit.length < 2 ? 0 : parseInt(versionSplit[1], 0);
+			releaseVersion = versionSplit.length == 3 ? parseInt(versionSplit[2], 0) : 0;
+		} else {
```

Comment:
I'd maybe wrap this whole block in try-catch, as it uses some array indexing without length checking. It shouldn't trigger often, but seeing some vendors come up with "new versioning schemes", it would be probably better to use default fail-save values for malformed string instead of just crashing the whole application. Logging which version string crashed it would also help when adding support for it and debugging.

Otherwise I really like it!